presto-blackhole/src/test/java/com/facebook/presto/plugin/blackhole/TestBlackHoleSmoke.java

```diff
@@ -226,6 +227,47 @@ public void pageProcessingDelay()
         assertThatQueryReturnsValue("DROP TABLE nation", true);
     }
 
+    @Test
+    public void testAllTypes()
+            throws Exception
+    {
+        assertThatQueryReturnsValue(
+                format("CREATE TABLE blackhole_all_types (" +
+                                "  _varchar VARCHAR(10)" +
+                                ", _bigint BIGINT" +
+                                ", _integer INTEGER" +
+                                ", _smallint SMALLINT" +
+                                ", _tinyint TINYINT" +
+                                ", _double DOUBLE" +
+                                ", _boolean BOOLEAN" +
+                                ", _decimal_short DECIMAL(3,2)" +
+                                ", _decimal_long DECIMAL(30,10)" +
+                                ") WITH ( %s = 1, %s = 1, %s = 1 ) ",
+                        ROWS_PER_PAGE_PROPERTY,
+                        PAGES_PER_SPLIT_PROPERTY,
+                        SPLIT_COUNT_PROPERTY),
+                true);
+
+        assertThatQueryReturnsValue("INSERT INTO blackhole_all_types VALUES (" +
+                "'abc', BIGINT '1', INTEGER '2', SMALLINT '3', TINYINT '4', DOUBLE '5', true, DECIMAL '3.14', DECIMAL '1234567890.123456789')", 1L);
+
+        MaterializedResult rows = queryRunner.execute("SELECT * FROM blackhole_all_types");
+        assertEquals(rows.getRowCount(), 1);
```

Comment:
it is misleading. You inserted a row and then selected completely different one. I would split this test into two. One with insert and one with select. What do you think?